Sync/Synchronizers/ClientsSynchronizer.swift

```diff
@@ -346,3 +345,31 @@ public class ClientsSynchronizer: TimestampedSingleCollectionSynchronizer, Synch
             >>> { deferMaybe(.Completed) }
     }
 }
+
+private extension UIDevice {
+    class func formFactorString() -> String {
+        var formfactor: String
+        let FORMFACTOR_PHONE = "phone"
+        let FORMFACTOR_LARGETABLET = "largetablet"
+        let FORMFACTOR_SMALLTABLET = "smalltablet"
+        
+        var size: Int = 2
+        sysctlbyname("hw.machine", nil, &size, nil, 0)
+        var machine = [CChar](count: Int(size), repeatedValue: 0)
+        sysctlbyname("hw.machine", &machine, &size, nil, 0)
+        let hardware: String = String.fromCString(machine)!
```

Comment:
Nit: I'm always a bit wary of explicit unwrap like this. Probably safer to use a `guard` and optional unwrap:

```swift
guard let hardware = String.fromCString(machine) else {
    return FORMFACTOR_PHONE
}
```